### ISSUE TITLE
When is the pytorch version available？

### ISSUE THREAD

**Issue by benjiachong** [aamini/evidential-deep-learning#2]
Hi, it is a very interesting and valuable work. I have two questions:
1.When the pytorch version is available?  
2.I notice the regression problem is used in the paper. What about the classification problem? Is it suitable for the classification problem? 
Thanks.

**Comment by aamini** [comment]
Thank you for your interest in the work!

1. It is in active development, but I imagine should be done soon. @Dariusrussellkish has already submitted a PR drafting a pytorch version as part of this repository that we will be merging into master ASAP. Please see #1 for more details. TLDR: PyTorch backend to this same repo will be available very soon!

2. Yes! Evidential learning is also possible for classification as well [[ref](https://arxiv.org/pdf/1806.01768.pdf)]. We provided the layers and loss functions to handle classification in this repo (please see code snippets below for pointers to the appropriate layers and losses). Since our work was focused on regression, the examples in the repo are also in this domain -- we have some code to also use the classification portion of the codebase to solve tasks like MNIST, CIFAR10, for example but have not pushed these. If you would like to contribute some classification examples using the existing layers and losses, we would be very happy to review and merge in! 

Classification layers: https://github.com/aamini/evidential-deep-learning/blob/7a22a2c8f35f5a2ec18fd37068b747935ff85376/evidential_deep_learning/layers/dense.py#L54-L68

Classification loss: https://github.com/aamini/evidential-deep-learning/blob/main/evidential_deep_learning/losses/discrete.py#L5-L32

**Comment by birolkuyumcu** [comment]
Dirichlet_SOS is not a loss function

**Comment by Ali-799** [comment]
> 
> 
> Thank you for your interest in the work!
> 
>     1. It is in active development, but I imagine should be done soon. @Dariusrussellkish has already submitted a PR drafting a pytorch version as part of this repository that we will be merging into master ASAP. Please see [Pytorch implementation #1](https://github.com/aamini/evidential-deep-learning/pull/1) for more details. TLDR: PyTorch backend to this same repo will be available very soon!
> 
>     2. Yes! Evidential learning is also possible for classification as well [[ref](https://arxiv.org/pdf/1806.01768.pdf)]. We provided the layers and loss functions to handle classification in this repo (please see code snippets below for pointers to the appropriate layers and losses). Since our work was focused on regression, the examples in the repo are also in this domain -- we have some code to also use the classification portion of the codebase to solve tasks like MNIST, CIFAR10, for example but have not pushed these. If you would like to contribute some classification examples using the existing layers and losses, we would be very happy to review and merge in!
> 
> 
> Classification layers:
> 
> https://github.com/aamini/evidential-deep-learning/blob/7a22a2c8f35f5a2ec18fd37068b747935ff85376/evidential_deep_learning/layers/dense.py#L54-L68
> 
> Classification loss: https://github.com/aamini/evidential-deep-learning/blob/main/evidential_deep_learning/losses/discrete.py#L5-L32

Hi @aamini ,
Thanks for sharing your valuable work with the community. Refer to your comment, i am working on a classification task and using the classfication layers and loss mentioned by you, for evidential learning. However, i faced errors while training with the Dirichlet_SOS loss. I troubleshooted and here is what solved my issue:
1. 1st is to remove parameter 't' from def Dirichlet_SOS(y, alpha, t). This isn't used anywhere in the loss function and calling the function without this parameter in model.compile, gives you an error.
2.  2nd is to add following line of code in the loss function before 'S = tf.reduce_sum(alpha, axis=1, keepdims=True)':
 alpha,prob =tf.split(alpha,2,axis=-1)
Without it, the dimensions of 'y' and 'm' don't match.
Hoping that you can verify my observations? Thanks
